fedimint-server/src/net/api.rs

```diff
@@ -141,95 +135,65 @@ impl ConsensusApi {
                     &mut dbtx.with_module_prefix(output.module_instance_id()),
                     output,
                 )
-                .await
-                .map_err(|e| TransactionSubmissionError::ModuleError(tx_hash, e))?;
+                .await?;
             funding_verifier.add_output(amount);
         }
 
         funding_verifier.verify_funding()?;
 
         self.api_sender
             .send(ApiEvent::Transaction(transaction))
-            .await
-            .map_err(|_e| TransactionSubmissionError::TxChannelError)?;
+            .await?;
+
         Ok(())
     }
 
-    pub async fn transaction_status(
-        &self,
-        txid: TransactionId,
-    ) -> Option<crate::outcome::TransactionStatus> {
+    pub async fn transaction_status(&self, txid: TransactionId) -> Option<TransactionStatus> {
         let mut dbtx = self.db.begin_transaction().await;
 
-        let accepted: Option<AcceptedTransaction> =
-            dbtx.get_value(&AcceptedTransactionKey(txid)).await;
+        let module_ids = dbtx.get_value(&AcceptedTransactionKey(txid)).await?;
 
-        if let Some(accepted) = accepted {
-            return Some(
-                self.accepted_transaction_status(txid, accepted, &mut dbtx)
-                    .await,
-            );
-        }
-
-        let rejected: Option<String> = self
-            .db
-            .begin_transaction()
-            .await
-            .get_value(&RejectedTransactionKey(txid))
+        let status = self
+            .accepted_transaction_status(txid, module_ids, &mut dbtx)
             .await;
 
-        if let Some(message) = rejected {
-            return Some(TransactionStatus::Rejected(message));
-        }
-
-        None
+        Some(status)
     }
 
-    pub async fn wait_transaction_status(
-        &self,
-        txid: TransactionId,
-    ) -> crate::outcome::TransactionStatus {
-        let accepted_key = AcceptedTransactionKey(txid);
-        let rejected_key = RejectedTransactionKey(txid);
-        tokio::select! {
-            (accepted, mut dbtx) = self.db.wait_key_check(&accepted_key, std::convert::identity) => {
-                self.accepted_transaction_status(txid, accepted, &mut dbtx).await
-            }
-            rejected = self.db.wait_key_exists(&rejected_key) => {
-                TransactionStatus::Rejected(rejected)
-            }
-        }
+    pub async fn wait_transaction_status(&self, txid: TransactionId) -> TransactionStatus {
+        let (outputs, mut dbtx) = self
+            .db
+            .wait_key_check(&AcceptedTransactionKey(txid), std::convert::identity)
+            .await;
+
+        self.accepted_transaction_status(txid, outputs, &mut dbtx)
+            .await
```

Comment:
This is a huge semantic change, we need to make sure the client still works in error cases. It's likely some state machines need to be changed.